src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Writer/CborWriter.Simple.cs

```diff
@@ -7,6 +7,10 @@ namespace System.Formats.Cbor
 {
     public partial class CborWriter
     {
+        // Canonical NaN representations as per RFC 7049
+        private static readonly float CanonicalNaNSingle = CborHelpers.Int32BitsToSingle(0x7fc00000);
+        private static readonly double CanonicalNaNDouble = BitConverter.Int64BitsToDouble(0x7ff8000000000000);
```

Comment:
nit: I think it would be better to simply calls these what they actually are, which is `PositiveQNaN`.

The concern being that they are canonical in the context of CBOR, but not more generally, and someone may get the wrong idea especially if just generally searching for things related to NaN in the repo.

Calling it what it is, a `+QNaN` and then leaving a comment that it's used because `RFC 7049` expects it as it's own canonical representation avoids the ambiguity, while potentially providing additional clarity and allowing us to clearly indicate why we don't use `float.NaN`

Reply:
OK